music_assistant/providers/internet-archive/parsers.py

```diff
@@ -0,0 +1,246 @@
+"""Metadata parsing utilities for the Internet Archive provider."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from typing import Any
+
+from music_assistant_models.enums import AlbumType, ImageType
+from music_assistant_models.media_items import (
+    Album,
+    Artist,
+    Audiobook,
+    MediaItemImage,
+    ProviderMapping,
+    Track,
+)
+from music_assistant_models.unique_list import UniqueList
+
+# from music_assistant.helpers import infer_album_type
+from .constants import AUDIOBOOK_COLLECTIONS
+from .helpers import clean_text, extract_year, get_image_url
+
+
+class InternetArchiveParsers:
```

Comment:
Not so sure if this needs to be in his own class.
i would rather make this into standalone helper functions - if you really rely on the provider just pass in the provider instance as argument but in general the parsers should be limited to convert raw data into a MA model